File: esop2020/benchmarks/seahorn/5-linger-dec/linger-dec-10-0.c
#include "seahorn/seahorn.h"

extern int nd();

void linger_dec_bound(int n, int *pa) {
  if (n == 0) return;
  *pa -= 1;
  int x = nd();
  linger_dec_bound(n - 1, nd() ? &x : pa);
}

int main() {
  int n = nd(), a = nd();
  int old_a = a;
  linger_dec_bound(n, &a);
  sassert(old_a >= a && old_a - a <= n);
}

File: esop2020/benchmarks/rust-horn/5-linger-dec/linger-dec-10-0.rs
pub fn rand<T>() -> T { loop {} }

fn linger_dec_bound(n: i32, ma: &mut i32) {
  if n == 0 {
    return;
  }
  *ma -= 1;
  let mut x = rand();
  linger_dec_bound(n - 1, if rand() { &mut x } else { ma });
}
fn main() {
  let n = rand();
  let mut a = rand();
  let old_a = a;
  linger_dec_bound(n, &mut a);
  assert!(old_a >= a && old_a - a <= n);
}
